### PR TITLE
Fix attachment lost issue

### PR DESCRIPTION
**Pull request by topikachu**
When a test both output "[[ATTACHMENT|/file]]" and has /target/surefire-reports/Test folder,
the attachments of the test cases are lost.
This happens when using TestNG.
Fix the issue to keep the previous ones.

### CODE REVIEW

**Comment by topikachu** [comment]
@jglick Would you please take a look at this change? The current plugin doesn't work with testNG well because of the issue.
Thanks.

**Comment by topikachu** [comment]
@pjhampton Add test in the new change.

File: src/test/java/hudson/plugins/junitattachments/AttachmentPublisherPipelineTest.java [hunk 1]
```diff
@@ -129,4 +129,32 @@ private String fileContentsFromResources(String fileName) throws IOException {
 
     }
 
+    @Test
+    public void testBothWellKnownFilenamesAndPatternAreAttached() throws Exception {
```
Comment (pjhampton) [inline src/test/java/hudson/plugins/junitattachments/AttachmentPublisherPipelineTest.java]: Ok, awesome. Thanks @topikachu 

One last thing, can you move this method below `annotationDoesNotFailForPipeline()` instead to keep it consistent.

**Comment by pjhampton** [comment]
Think we are good after that small formatting change. Thanks for the contribution @topikachu

**Comment by dhc8** [comment]
This was merged back in January, but it doesn't look like a release was ever created. Any chance this could get released?

**Pull request merged**